### ISSUE TITLE
[GENERAL] Linebreaks don't work on writing prompts.

### ISSUE THREAD

**Issue by TheToaster98** [VOREStation/VOREStation#14047]
#### Brief description of the bug
When using the hotkeys or the verbs on the sidetab, linebreaks don't work.

#### What you expected to happen
For linebreaks to work on the writing prompts.

#### What actually happened
[you do like \n this](https://cdn.discordapp.com/attachments/719676195637297263/1038903231457271890/image.png)

#### Steps to reproduce
- Press T/5/6/Y or any other hotkey that opens a writing prompt. Or use the verbs on the side.
- Try to use a linebreak.
- Linebreak breaks and prints \n
- Type exactly the same on the command line.
- Works fine!

#### Code Revision
- Commit: [7f2c88c93b902b7421ab360360fdcdefb3ac0218](http://127.0.0.1:59065/tmp13996/%3Ca%20href=%22https://github.com/VOREStation/VOREStation/commit/7f2c88c93b902b7421ab360360fdcdefb3ac0218%22%3Ehttps://github.com/VOREStation/VOREStation/commit/7f2c88c93b902b7421ab360360fdcdefb3ac0218%3C/a%3E)
 

#### Anything else you may wish to add:
- Tested on both TGUI and normal Byond input boxes, neither works, only the command bar.

**Comment by KillianKirilenko** [comment]
This one's on me for not testing input boxes for say/etc. since I consistently forget that those exist. They must use a different sanitize proc or something.

**Comment by ItsSelis** [comment]
iirc the input boxes (at least the multiline ones) have linebreaks with the enter key (since enter puts a U+000A inside of there which really is just a so-called new line or line feed in UTF-8).
Though yeah, **normally** the input boxes do not have such functionality to sanitze \n, \t etc.

**Comment by KillianKirilenko** [comment]
> iirc the input boxes (at least the multiline ones) have linebreaks with the enter key (since enter puts a U+000A inside of there which really is just a so-called new line or line feed in UTF-8). Though yeah, **normally** the input boxes do not have such functionality to sanitze \n, \t etc.

Nope, hitting enter with either old-style or TGUI just sends the message. Tried shift-enter and ctrl-enter as well; those just send it too.

**Comment by ItsSelis** [comment]
> Nope, hitting enter with either old-style or TGUI just sends the message. Tried shift-enter and ctrl-enter as well; those just send it too. Alt-enter does nothing with old-style, or sends with TGUI.

I can try to port a fix from tg which had a similar issue with multiline tgui input boxes. Though the old ones would require sanitation that replaces the \n with an actual line break.

 Though, there should be limits of such sanitations too imo. Otherwise people could do many new lines (\n\n\n\n\n\n\n) in a row or cause other unintended behaviour accidentally (like inputting new lines into boxes that usually do not accept such: aka, only letting multiline input boxes sanitize \n or something).

**Comment by KillianKirilenko** [comment]
> I can try to port a fix from tg which had a similar issue with multiline tgui input boxes. Though the old ones would require sanitation that replaces the \n with an actual line break.
> 
> Though, there should be limits of such sanitations too imo. Otherwise people could do many new lines (\n\n\n\n\n\n\n) in a row or cause other unintended behaviour accidentally (like inputting new lines into boxes that usually do not accept such: aka, only letting multiline input boxes sanitize \n or something).

I can pretty much guarantee you that people would appreciate it, at least judging by the reaction my initial attempt (#14017) got.

Currently you're limited to six linebreaks, which means four paragraphs (initial + 3 extra) total. Any more than that and it strips them out again, so it gets obvious if you're trying to abuse the chatbox to be disruptive.

**Comment by ItsSelis** [comment]
> Some more information I have gotten from some testing:
> 
>     * In writing prompts by the verbs, or those without TGUI, pressing enter will make a linebreak... but only if they are multiline.
> 
>     * You can't press enter on a TGUI prompt, since it will just send the message, shift, ctrl and Alt-enter send the message too, but if you paste a text with linebreaks from somewhere else it will have those linebreaks.
> 
>     * Someone said that "Doing a \ in the command line deletes everything after it" but I have been unable to replicate it.

tgui_input_text accepts `prevent_enter ` as a boolean arguemnt (which defaults to FALSE). Making that true supresses the send on enter functionality. 

For example, the records in the character creator have that:
![grafik](https://user-images.githubusercontent.com/12716288/200308712-1e9c912f-50cd-43c3-bcdb-3293e3949b19.png)

**Comment by Heroman3003** [comment]
Is this still an issue?

**Comment by TheToaster98** [comment]
I tested it as of right now on this very moment, it still is an issue, yes.
